src/Akka.Cluster.Hosting.Tests/ClusterSingletonSpecs.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Event;
+using Akka.Hosting;
+using Akka.Remote.Hosting;
+using Akka.TestKit.Xunit2.Internals;
+using FluentAssertions;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Akka.Cluster.Hosting.Tests;
+
+public class ClusterSingletonSpecs
+{
+    public ClusterSingletonSpecs(ITestOutputHelper output)
+    {
+        Output = output;
+    }
+
+    public ITestOutputHelper Output { get; }
+    
+    private class MySingletonActor : ReceiveActor
+    {
+        public static Props MyProps => Props.Create(() => new MySingletonActor());
+
+        public MySingletonActor()
+        {
+            ReceiveAny(_ => Sender.Tell(_));
+        }
+    }
+
+    private async Task<IHost> CreateHost(Action<AkkaConfigurationBuilder> specBuilder, ClusterOptions options)
+    {
+        var tcs = new TaskCompletionSource();
+        using var cancellationTokenSource = new CancellationTokenSource(TimeSpan.FromSeconds(10));
+
+        var host = new HostBuilder()
+            .ConfigureServices(collection =>
+            {
+                collection.AddAkka("TestSys", (configurationBuilder, provider) =>
+                {
+                    configurationBuilder
+                        .WithRemoting("localhost", 0)
+                        .WithClustering(options)
+                        .WithActors((system, registry) =>
+                        {
+                            var extSystem = (ExtendedActorSystem)system;
+                            var logger = extSystem.SystemActorOf(Props.Create(() => new TestOutputLogger(Output)), "log-test");
+                            logger.Tell(new InitializeLogger(system.EventStream));
+                        })
+                        .WithActors(async (system, registry) =>
+                        {
+                            var cluster = Cluster.Get(system);
+                            cluster.RegisterOnMemberUp(() =>
+                            {
+                                tcs.SetResult();
+                            });  
+                            if (options.SeedNodes == null || options.SeedNodes.Length == 0)
```

Comment:
If no seed nodes are provided to `ClusterOptions`, assume we're the seed node and self-join.